src/connection.cc

```diff
@@ -751,13 +751,7 @@ char* Connection::NewCString(v8::Local<v8::Value> val) {
   Nan::HandleScope scope;
 
   v8::Local<v8::String> str = Nan::To<v8::String>(val).ToLocalChecked();
-
-  #if defined(V8_MAJOR_VERSION) && V8_MAJOR_VERSION >= 7
-    int len = str->Utf8Length(v8::Isolate::GetCurrent()) + 1;
-  #else
-    int len = str->Utf8Length() + 1;
-  #endif
-
+  int len = Nan::Utf8String(str).length() + 1;
```

Comment:
I thing you can avoid the `#ifdef` for `WriteUtf8()` by dereferencing `Nan::Utf8String`; e.g. like this (`Nan::To<String>` is also not needed then):
```
Nan::Utf8String nanStr = Nan::Utf8String(val);
char* buffer = new char[nanStr.length()];
strcpy(char, *nanStr);
```

Reply:
When using this:
```cpp
char* Connection::NewCString(v8::Local<v8::Value> val) {
  Nan::HandleScope scope;

  Nan::Utf8String nanStr = Nan::Utf8String(val);
  char* buffer = new char[nanStr.length() + 1];
  strcpy(buffer, *nanStr);

  return buffer;
}
```
I get this error:
```sh
../src/connection.cc:753:47: error: use of deleted function ‘Nan::Utf8String::Utf8String(Nan::Utf8String&&)’
   Nan::Utf8String nanStr = Nan::Utf8String(val);
                                               ^
```

Reply:
Be careful in replacing `new[]` with `malloc`. Memory allocated with `new[]` has to be released by `delete[]` whereas `malloc` matches with `free`. On several platforms it works by luck if it is mixed but actually it's undefined behaviour.

Reply:
I followed your advice and changed this to use new char[] instead ^^